tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -330,15 +319,18 @@ TEST_CASE("Test Transformer ranking") {
 
             pgm_tap::TrafoGraphEdgeProperties const regulated_edge_weights = get_edge_weights(g);
             pgm_tap::TrafoGraphEdgeProperties const ref_regulated_edge_weights{
-                {{0, 1}, 0}, {{1, 2}, 1}, {{2, 3}, 2}, {{3, 4}, 1}, {{4, 5}, 0}};
+                {{0, 1}, 1}, {{1, 2}, 2}, {{2, 3}, 2}, {{3, 4}, 1}};
             CHECK(regulated_edge_weights == ref_regulated_edge_weights);
         }
 
         SUBCASE("Ranking complete the graph") {
-            pgm_tap::RankedTransformerGroups order = pgm_tap::rank_transformers(state);
-            pgm_tap::RankedTransformerGroups const ref_order{
-                {Idx2D{3, 0}, Idx2D{3, 1}, Idx2D{4, 0}, Idx2D{3, 3}, Idx2D{3, 2}, Idx2D{3, 4}}};
-            CHECK(order == ref_order);
+            // (TODO: jguo) existing demo grid is not compatible with the updated ranking
+            // The test grid needs to be updated here to match the new ranking logic
+            // pgm_tap::RankedTransformerGroups order = pgm_tap::rank_transformers(state);
+            // pgm_tap::RankedTransformerGroups const ref_order{
+            //     {{Idx2D{3, 0}, Idx2D{3, 1}, Idx2D{4, 0}}, {Idx2D{3, 3}, Idx2D{3, 2}, Idx2D{3, 4}}}};
+            // CHECK(order == ref_order);
```

Comment:
New unit test need to be added in the following up PR to give a valid grid configuration. I.e., re-work the OneNote grid and make it correct.

Reply:
Confirmed. Left for the follow up validation PR. NB, that PR is mainly about the validation, but also about correcting the unit test grid.